### PR TITLE
Add middleware to App Gateway

### PR DESCRIPTION
**Pull request by derekadombek**
## Description
Forgot to add this on my previous PR

this should hopefully allow all the services to communicate properly 🤞 

## Related Issues
[Link any related issues or tasks from your project management system.]

## Checklist
- [x] The title of this PR is descriptive and concise.
- [x] My changes follow the style guidelines of this project.
- [x] I have added or updated test cases to cover my changes.
- [x] I've let the team know about this PR by linking it in the review channel

### CODE REVIEW

File: .github/workflows/build-deploy-ocr.yml [hunk 1]
```diff
@@ -54,14 +56,14 @@ jobs:
...
-          docker-context-path: ./OCR/
+          dockerfile-path: ./backend/Dockerfile
+          docker-context-path: ./backend/
           api-name: ocr
 
   deploy-ocr:
     name: Deploy OCR
     runs-on: ubuntu-latest
-    environment: ${{ inputs.deploy-env }}
+    environment: dev
```
Comment (marycrawford) [inline .github/workflows/build-deploy-ocr.yml:66]: Do we want to hardcode `dev` here?  Or can we make dynamic so if whomever picks it up can select `dev`, as well as, `stage` or `prod` (after they build out those environments)?

File: ops/terraform/modules/app_gateway/main.tf [hunk 2]
```diff
@@ -142,33 +187,48 @@ resource "azurerm_application_gateway" "load_balancer" {
...
   url_path_map {
     name                               = "${var.name}-${var.env}-urlmap"
     default_backend_address_pool_name  = local.backend_address_pool_name_static
     default_backend_http_settings_name = local.http_setting_name_static
-    default_rewrite_rule_set_name      = "${var.name}-routing"
+    default_rewrite_rule_set_name      = "${var.name}-middleware-routing"
 
     path_rule {
-      name                       = "api"
+      name                       = "ocr"
```
Comment (marycrawford) [inline ops/terraform/modules/app_gateway/main.tf:214]: Is the name just `ocr` or `api_ocr`?

File: .github/workflows/build-deploy-ocr.yml (hunk 1 above)
Comment (derekadombek) [reply .github/workflows/build-deploy-ocr.yml:66]: this is how we do local github action testing. I will make this a draft

File: ops/terraform/modules/app_gateway/main.tf (hunk 2 above)
Comment (derekadombek) [reply ops/terraform/modules/app_gateway/main.tf:214]: oh the name here doesn't really matter, just needs to be different than the middleware rule.

File: .github/workflows/build-deploy-ocr.yml (hunk 1 above)
Comment (derekadombek) [reply .github/workflows/build-deploy-ocr.yml:66]: also wouldnt need to like this if we create a seperate worflow for building and deploying the middleware

**Review by marycrawford — APPROVED** [review]

**Pull request merged**